### PR TITLE
Update zlib to 1.2.13

### PR DESCRIPTION
**Pull request by cscd98**
## Description
Updates zlib to 1.2.13.

## Motivation and context
Includes bug fixes and a security fix - https://nvd.nist.gov/vuln/detail/CVE-2022-37434

## How has this been tested?
Compilation only at this stage. Unable to test front end at present.

Will require https://github.com/madler/zlib/releases/download/v1.2.13/zlib-1.2.13.tar.xz to be uploaded to http://mirrors.kodi.tv/build-deps/sources/

## What is the effect on users?
None

## Types of change

- [x] **Improvement** (non-breaking change which improves existing functionality)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/zlib/ZLIB-VERSION [hunk 1]
```diff
@@ -1,5 +1,5 @@
 LIBNAME=zlib
-VERSION=1.2.12
-ARCHIVE=$(LIBNAME)-$(VERSION).tar.xz
-SHA512=12940e81e988f7661da52fa20bdc333314ae86a621fdb748804a20840b065a1d6d984430f2d41f3a057de0effc6ff9bcf42f9ee9510b88219085f59cbbd082bd
+VERSION=1.2.13
+ARCHIVE=$(LIBNAME)-$(VERSION).tar.gz
```
Comment (fuzzard) [inline tools/depends/target/zlib/ZLIB-VERSION:3]: stick to the xz, better compression. same with native

Comment (fuzzard) [reply tools/depends/target/zlib/ZLIB-VERSION:3]: Also for testing, you can use BASE_URL to provide an alternate to kodi mirrors.

```
BASE_URL=https://www.zlib.net
```

will use the url https://www.zlib.net/zlib-1.2.13.tar.xz

**Comment by craigcarnell** [comment]
@fuzzard Updated PR to reflect xz

**Comment by fuzzard** [comment]
Can you squash this in a single commit.

Have uploaded zlib-1.2.13.tar.xz to mirrors just now, so should become available shortly

**Comment by craigcarnell** [comment]
@fuzzard Done

**Pull request merged**